families-api/infra/__main__.py

```diff
@@ -0,0 +1,165 @@
+                }
+            ],
+        }
+    ),
+)
+
+# Allow access to specific SSM Parameter Store secrets
+families_api_ssm_policy = aws.iam.RolePolicy(
+    "families-api-instance-role-ssm-policy",
+    role=families_api_instance_role.id,
+    policy=json.dumps(
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": ["ssm:GetParameters"],
+                    "Resource": [
+                        f"arn:aws:ssm:eu-west-1:{account_id}:parameter/families-api/apprunner/*"
+                    ],
+                }
+            ],
+        }
+    ),
+)
+
+families_api_apprunner_navigator_database_url = aws.ssm.Parameter(
+    "families-api-apprunner-navigator-database-url",
+    name=generate_secret_key("families-api", "apprunner", "NAVIGATOR_DATABASE_URL"),
+    description="The URL string to connect to the navigator database",
+    type=aws.ssm.ParameterType.SECURE_STRING,
+    # This value is managed directly in SSM
+    value="PLACEHOLDER",
+    opts=pulumi.ResourceOptions(
+        # This value is managed directly in SSM
+        ignore_changes=["value"]
+    ),
+)
+
+families_api_ecr_repository = aws.ecr.Repository(
+    "families-api-ecr-repository",
+    encryption_configurations=[
+        {
+            "encryption_type": "AES256",
+        }
+    ],
+    image_scanning_configuration={
+        "scan_on_push": False,
+    },
+    image_tag_mutability="MUTABLE",
+    name="families-api",
+    opts=pulumi.ResourceOptions(protect=True),
+)
+
+
+families_api_apprunner_service = aws.apprunner.Service(
+    "families-api-apprunner-service",
+    auto_scaling_configuration_arn=f"arn:aws:apprunner:eu-west-1:{account_id}:autoscalingconfiguration/DefaultConfiguration/1/00000000000000000000000000000001",
+    health_check_configuration={
+        "interval": 10,
+        "protocol": "TCP",
+        "timeout": 5,
+    },
+    instance_configuration={
+        "instance_role_arn": families_api_instance_role.arn,
+    },
+    network_configuration={
+        "egress_configuration": {
+            "egress_type": "VPC",
+            # This is only needed because we have hidden the RDS store in a different VPC to all our other resources
+            "vpc_connector_arn": apprunner_vpc_connector_arn,
+        },
+        "ingress_configuration": {
+            "is_publicly_accessible": True,
+        },
+        "ip_address_type": "IPV4",
+    },
+    observability_configuration={
+        "observability_enabled": False,
+    },
+    service_name="families-api",
+    source_configuration={
+        "authentication_configuration": {
+            "access_role_arn": families_api_role.arn,
+        },
+        "image_repository": {
+            "image_configuration": {
+                "runtime_environment_secrets": {
+                    "NAVIGATOR_DATABASE_URL": families_api_apprunner_navigator_database_url.arn,
```

Comment:
@katybaulch might be of interest - using the secure pattern of storing values encrypted in SSM and then referencing them in the App.

Reply:
@henry-cpr you mentioned this in a previous chat too, moving away from Pulumi config.